Lib/asyncio/tasks.py

```diff
@@ -455,11 +457,9 @@ async def wait_for(fut, timeout):
             # exception, we should re-raise it
             # See https://bugs.python.org/issue40607
             try:
-                fut.result()
+                return fut.result()
```

Comment:
So, this fixes that original issue, by returning the result instead of ignoring it and raising a timeout.

Reply:
There is also some identical looking code at line 420. So, maybe that should also return. I've reached my limit for the moment, so maybe someone else can figure that out.